src/object/cli_obj.c

```diff
@@ -832,7 +815,7 @@ obj_dkey2grpidx(struct dc_object *obj, uint64_t hash, unsigned int map_ver)
 	D_ASSERT(grp_size > 0);
 
 	D_RWLOCK_RDLOCK(&obj->cob_lock);
-	if (obj->cob_version != map_ver || map_ver < pool_map_ver) {
+	if (obj->cob_version != pool_map_ver) {
```

Comment:
is this change intended?

looks one difference is for example the pool map refreshed and obj layout refreshed by another thread (so both obj->cob_version and pool_map_ver refreshed), but this thread's IO handling with stale map_ver, this case may need return ESTALE as previous code?